server.go

```diff
@@ -740,6 +740,17 @@ func (server *Server) loadConfig(configurations configs, globalConfiguration Glo
 								negroni.Use(metricsMiddlewareBackend)
 							}
 						}
+
+						if len(frontend.WhitelistSourceRange) > 0 {
+							log.Info("Configured IP Whitelists: ", frontend.WhitelistSourceRange)
+							ipSourceRanges := frontend.WhitelistSourceRange
+							ipWhitelistMiddleware, err := middlewares.NewIPWhitelister(ipSourceRanges)
+							if err != nil {
+								log.Fatal("Error creating IP Whitelister: ", err)
+							}
+							negroni.Use(ipWhitelistMiddleware)
+						}
+
```

Comment:
This part needs to be covered by a test. I know it's tricky because we have zero tests yet, but we need to start somewhere.

For a starter, [here are my tests from a different PR](https://github.com/containous/traefik/blob/539e22a37fbe28f4f7f93e5825960fee7a714990/server_test.go).